backend/Buk.UniversalGames.Data/Repositories/GameDataRepository.cs

```diff
@@ -61,7 +61,7 @@ public async Task<List<MatchListItem>> GetMatches(int leagueId, int? gameId = nu
                         Team2Id = match.Team2Id,
                         Team2 = match.Team2.Name,
                         WinnerId = match.WinnerId.GetValueOrDefault(),
-                        Winner = match.WinnerId.HasValue ? (match.WinnerId.Value == match.Team1Id ? match.Team1.Name : match.Team2.Name) : "",
+                        Winner = match.WinnerId.HasValue ? (match.WinnerId!.Value == match.Team1Id ? match.Team1.Name : match.Team2.Name) : "",
```

Comment:
I think such complicated logic statements can benefit from modern C#'s switch expression:

```suggestion
                        Winner = match.WinnerId switch {
                          null => "",
                          var teamId when teamId == match.Team1Id => match.Team1.Name,
                          var teamId when teamId == match.Team2Id => match.Team2.Name,
                        }
```
> **Warning**
> I haven't tested the code above, just wrote it by heart.

Reply:
Nice idea, aside from the lengthy matching epressions it would make stuff more explicit. Caveat here is that switch cannot be used like this in an expression tree, so this one won't fly unfortunately

Reply:
Ah, true, I'd missed that. It's a shame really, because for this case it would have worked, but the general cases can be too complicated for expression trees.